modules/services/nix-gc.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix.gc;
+
+  mkCalendarInterval = frequency:
```

Comment:
@rycee I have taken the liberty to create a function that provides a mapping from systemd time-frequency enums to that of launchd’s. is it a good idea to also expose an option to let override the frequency, if they don’t want to use the default mapping? How would you think that should be? separate option for both macOS and linux frequency?

Reply:
My spontaneous opinion would be to have a single duration type that can be mapped into a systemd or launchd interval. I've made such years ago. I've looked at this a few times and eventually I think it would be good to introduce. See e.g. https://github.com/NixOS/nixpkgs/pull/43852 for my, perhaps, most elaborate take on this.

I think I have some half-baked simplified version of that as well that probably could be resurrected.

In any case, perhaps it would be sufficient to have what you have now and revisit in the future?